src/main/java/org/spongepowered/common/mixin/core/world/chunk/ChunkMixin.java

```diff
@@ -545,7 +545,7 @@ public IBlockState setBlockState(final BlockPos pos, final IBlockState state) {
             // Sponge End
             TileEntity tileentity = this.getTileEntity(pos, net.minecraft.world.chunk.Chunk.EnumCreateEntityType.CHECK);
 
-            if (tileentity == null) {
+            if (tileentity == null || (transaction != null && transaction.queuedRemoval != null)) {
```

Comment:
Wouldn't it just make more sense to check if `existing != null`? Or are you using the transaction as a check if we're capturing in the first place before checking if the capture had captured a tile entity to be removed?

Reply:
We capture the removal here [L434](https://github.com/ginyai/SpongeCommon/blob/8f87754fcf62686b167cdf96dd5c23c7883adbea/src/main/java/org/spongepowered/common/mixin/core/world/chunk/ChunkMixin.java#L434).
Affect the selection of this if expression

Reply:
Ok so it's bundling the combined check that required the tile entity being queued for removal in the first place. Can you comment that in the change then (since `tileentity == null` is the only vanilla check, by adding the transaction check it's now sponge modified). I try to keep this method commented about its changes compared to the original code to avoid confusion between updates and when reading the method to see where sponge is making a change compared to the original engine code.

Reply:
@gabizou like this?
```suggestion
            // Sponge Start - Additional check for the tile entity being queued for removal.
            // if (tileentity == null) { // Sponge
            if (tileentity == null || (transaction != null && transaction.queuedRemoval != null)) {
```

Reply:
More like 
```suggestion
            // if (tileentity == null) { // Vanilla
            // Sponge - we want to check for queued tile entity removals, as well as existing tile entities.
            if (tileentity == null || (transaction != null && transaction.queuedRemoval != null)) {
```